pkg/containers/containers.go

```diff
@@ -182,62 +200,87 @@ func (c *Containers) populate() error {
 // saving container information in Containers CgroupInfo map.
 // NOTE: ALL given cgroup dir paths are stored in CgroupInfo map.
 func (c *Containers) CgroupUpdate(cgroupId uint64, path string, ctime time.Time) (CgroupInfo, error) {
-	containerId, runtime := getContainerIdFromCgroup(path)
+	containerId, containerRuntime := getContainerIdFromCgroup(path)
+	container := runtime.ContainerMetadata{
+		ContainerId: containerId,
+	}
 
 	info := CgroupInfo{
-		Path:        path,
-		ContainerId: containerId,
-		Runtime:     runtime,
-		Ctime:       ctime,
+		Path:      path,
+		Container: container,
+		Runtime:   containerRuntime,
+		Ctime:     ctime,
 	}
 
 	c.mtx.Lock()
 	c.cgroups[uint32(cgroupId)] = info
 	c.mtx.Unlock()
 
+	c.enrichCgroupInfo(cgroupId)
+
 	return info, nil
 }
 
+func (c *Containers) enrichCgroupInfo(cgroupId uint64) {
```

Comment:
Could you add function documentation here and in most of other functions you created ? When yaniv created this code I did a PR on top just to document each function, as the cgroup id <-> container id mapping was not very straightforward to understand. Same happens here.

Reply:
Yes, i'll work on adding more documentation around this PR.

Reply:
Thanks. Please mark this done once you're finished with this.